pkg/reconciler/knativeeventing/knativeeventing.go

```diff
@@ -45,10 +45,11 @@ type Reconciler struct {
 	kubeClientSet kubernetes.Interface
 	// kubeClientSet allows us to talk to the k8s for operator APIs
 	operatorClientSet clientset.Interface
-	// config is the manifest of KnativeEventing
-	config mf.Manifest
-	// targetVersion is the version of the KnativeEventing manifest to install
-	targetVersion string
+	// manifest is empty, but with a valid client and logger. all
+	// manifests are immutable, and any created during reconcile are
+	// expected to be appended to this one, obviating the passing of
+	// client & logger
+	manifest mf.Manifest
```

Comment:
I don't think I like the semantics here. It feels backwards to create an empty manifest and then append to it and imo it's a sign of the manifestival API being wrong in this case.

Maybe they need a few tweaks? I feel like the client and the logger should only be passed to the side-effecting functions, like `Apply`, `Delete` etc and not be passed around with the manifest itself, which is (as you say) just an immutable collection of resources.

Reply:
Awkward, maybe. Wrong seems harsh. I suppose I could include client/logger as options to Apply/Delete/etc.

Personally, I liked having a single manifest member instead of both client and logger members. Having any members at all presents challenges as we introduce shared logic between the two reconcilers. That's what the `Parent` function addressed. I suppose the context parameter can be used for that, too.